From 5917f06a9a4bd37dc0496d235349e558787d6688 Mon Sep 17 00:00:00 2001
From: Ren Chenglei <chenglei.ren@intel.com>
Date: Fri, 9 Jun 2023 21:35:43 +0530
Subject: [PATCH] INTERNAL: Fix CTS CtsNativeHardwareTestCases failures

With allocator & mapper 4.0, there is some regression in
CTS CtsNativeHardwareTestCases. Compared with allocator &
mapper 2.0, we still not support two mipmap usage, and also
not support to allocate more than 1 layer.

Tracked-On: OAM-95612
Signed-off-by: Ren Chenglei <chenglei.ren@intel.com>
---
 cros_gralloc/gralloc4/CrosGralloc4Utils.cc | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/cros_gralloc/gralloc4/CrosGralloc4Utils.cc b/cros_gralloc/gralloc4/CrosGralloc4Utils.cc
index 3451b0b..f216c8e 100644
--- a/cros_gralloc/gralloc4/CrosGralloc4Utils.cc
+++ b/cros_gralloc/gralloc4/CrosGralloc4Utils.cc
@@ -55,6 +55,14 @@ int convertToDrmFormat(PixelFormat format, uint32_t* outDrmFormat) {
 }
 
 int convertToBufferUsage(uint64_t grallocUsage, uint64_t* outBufferUsage) {
+#ifdef USE_GRALLOC1
+    if ((grallocUsage & BufferUsage::GPU_MIPMAP_COMPLETE) ||
+            (grallocUsage & BufferUsage::GPU_CUBE_MAP)) {
+        ALOGE("GPU_MIPMAP_COMPLETE or GPU_CUBE_MAP not supported");
+        return -EINVAL;
+    }
+#endif
+
     *outBufferUsage = cros_gralloc_convert_usage(grallocUsage);
     return 0;
 }
@@ -108,6 +116,13 @@ int convertToCrosDescriptor(const BufferDescriptorInfo& descriptor,
         ALOGE("Failed to convert descriptor. Unsupported usage flags %s", usageString.c_str());
         return -EINVAL;
     }
+#ifdef USE_GRALLOC1
+    if (descriptor.layerCount > 1) {
+        ALOGE("Failed to convert descriptor. Can't support more than 1 layercount %d\n",
+                descriptor.layerCount);
+        return -EINVAL;
+    }
+#endif
     return 0;
 }
 
-- 
2.40.1

